### PR TITLE
Fix broken node_types api index action

### LINKED ISSUE #148: 500 error on wheelmap.org/api/categories/1/node_types
- I call http://wheelmap.org/api/categories/1/node_types?api_key=MYAPIKEY&per_page=6 in my browser where MYAPIKEY ist the key I got from my profile page
- I expect to see a result set.
- Instead, I get a 500 error. Other API request with my API key seem to work, for example http://wheelmap.org/api/nodes/3325161208?api_key=MYAPIKEY

### PR DESCRIPTION
**Pull request by schultyy**
Resolves #148 

This fixes `'NoMethodError: undefined method find' for Api::Category:Module`

Apparently since inherited_resources 1.6 the automatic resolving of class names in `belongs_to` calls is broken. We work around this by defining the `class_name` explicitly.
See https://github.com/josevalim/inherited_resources/pull/407 for more details.

### CODE REVIEW

**Comment by Hoverbear** [comment]
Is this expected?

screen shot 2017-01-12 at 14 55 13
screen shot 2017-01-12 at 14 56 23

**Comment by schultyy** [comment]
@Hoverbear No that's not expected. This occurs because the required monkey patch haven't been loaded before. Interestingly this does not occur in CI nor in production.

File: spec/controllers/api/node_types_controller_spec.rb [hunk 1]
```diff
@@ -13,12 +13,26 @@
 
     before :each do
       @category = FactoryGirl.create(:category)
-      FactoryGirl.create(:category, :identifier => 'education')
+      education_category = FactoryGirl.create(:category, :identifier => 'education')
       @node_type = FactoryGirl.create(:node_type, :category => @category)
+      education_node_type = FactoryGirl.create(:node_type, :category => education_category)
```
Comment (1000miles) [inline spec/controllers/api/node_types_controller_spec.rb]: Can you please remove the variable `education_node_type` since it is not called anywhere?

**Comment by 1000miles** [comment]
Deployed to prduction today with [release 20170116133241](https://github.com/sozialhelden/wheelmap/releases/tag/production-20170116133241).

Result for `api/categories/1/node_types?api_key=xxxxx`:

**Pull request merged**